plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -2892,6 +2893,129 @@ public void testUnregisterPartitionWithNullArgument()
+        assertQueryFails("CALL system.unregister_table('%s', '%s')".formatted(TPCH_SCHEMA, tableName), "Table .* not found");
+    }
+
+    @Test
+    public void testUnregisterView()
+    {
+        String viewName = "test_unregister_view_" + randomNameSuffix();
+        assertUpdate("CREATE VIEW %s AS SELECT * from region".formatted(viewName));
+        assertQuery("SELECT count(*) FROM " + viewName, "SELECT 5");
+
+        assertQueryFails("CALL system.unregister_table('%s', '%s')".formatted(TPCH_SCHEMA, viewName), "Not a Hive table '%s'".formatted(viewName));
+    }
+
+    @Test
+    public void testUnregisterExternalTable()
+    {
+        String tableName = "test_table" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s ( dummy_col bigint)".formatted(tableName));
+        assertUpdate(format("INSERT INTO %s (dummy_col) VALUES 1, 2, 3", tableName), 3);
+        assertQuery("SELECT count(*) FROM " + tableName, "SELECT 3");
+        String tableLocation = getTableLocation(tableName);
+        assertUpdate(format("CALL system.unregister_table('%s', '%s')", TPCH_SCHEMA, tableName));
+
+        String externalTableName = "test_unregister_external_table" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (dummy_col bigint)  with (external_location='%s')".formatted(externalTableName, tableLocation));
+        assertQuery("SELECT count(*) FROM " + externalTableName, "SELECT 3");
+
+        assertUpdate(format("CALL system.unregister_table('%s', '%s')", TPCH_SCHEMA, externalTableName));
+
+        assertUpdate("CREATE TABLE %s (dummy_col bigint)  with (external_location='%s')".formatted(externalTableName, tableLocation));
+        assertQuery("SELECT count(*) FROM " + externalTableName, "SELECT 3");
+
+        assertUpdate("DROP TABLE " + externalTableName);
+    }
+
+    @Test
+    public void testUnregisterPartitionedTable()
+    {
+        String tableName = "test_insert_empty_partitioned_table" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE %s (name VARCHAR, age INT, state VARCHAR) WITH (partitioned_by = ARRAY[ 'state' ]) ".formatted(tableName));
+        assertUpdate(format("INSERT INTO %s (name, age, state) VALUES ('Alice', 20, 'NY'), ('Bob', 30, 'CA'), ('Mike', 25, 'MA')", tableName), 3);
+        assertQuery("SELECT count(*) FROM " + tableName, "SELECT 3");
+        assertQuery(format("SELECT count(*) FROM \"%s$partitions\"", tableName), "SELECT 3");
+        String tableLocation = getPartitionedTableLocation(tableName, 1);
+
+        assertUpdate(format("CALL system.unregister_table('%s', '%s')", TPCH_SCHEMA, tableName));
+        assertQueryFails("SELECT count(*) FROM " + tableName, ".*%s' does not exist".formatted(tableName));
+
+        assertUpdate("CREATE TABLE %s (name VARCHAR, age INT, state VARCHAR) WITH (partitioned_by = ARRAY[ 'state' ],external_location='%s')".formatted(tableName, tableLocation));
+        assertQuery("SELECT count(*) FROM " + tableName, "SELECT 3");
```

Comment:
assertQuery("SELECT count(*) FROM " + tableName, "VALUES 3");

Reply:
I'd actually select here all the values that are expected to be on the table.

Reply:
The 3010 line does not seem to have that kind assertion.